### ISSUE TITLE
Docker build exiting early after step 2 (independent of what the step is doing)

### ISSUE THREAD

**Issue by ssolari** [docker/for-mac#2667]
Upgrading to 17.12.0-ce introduced this problem.  Confounding the problem is that I did some image cleanup (simply deleting non used images).   

### Expected behavior

Docker runs all steps in Dockerfile.

### Actual behavior

The docker build is simply exiting after step 2.   When we comment out step 2, it just exits at the next step. 

### Information

Diagnostic ID: 04C8A9AD-0229-483D-BF96-31AFF9BB5DBD

### Steps to reproduce the behavior

On any images on our computer, the following exits after step 2.  If I replace with a new image (that did not exist on the computer) like 'alpine' then it works.   However, it is reproducible with multiple images I tried that were on my computer before the upgrade. 
```
FROM continuumio/miniconda3
ADD http://www.random.org/strings/?num=10&len=8&digits=on&upperalpha=on&loweralpha=on&unique=on&format=plain&rnd=new uuid
ADD http://www.random.org/strings/?num=10&len=8&digits=on&upperalpha=on&loweralpha=on&unique=on&format=plain&rnd=new uuid
```

**Comment by ssolari** [comment]
An uninstall and complete reinstall of docker (which deleted all images) seesm to have fixed the issue.

**Comment by akimd** [comment]
Hi!

I can't reproduce your problem.  Please try with the most recent version of Docker (https://download.docker.com/mac/edge/23022/Docker.dmg), and create a new issue if you have the problem again.

Thanks!
